CHANGELOG.md

```diff
@@ -5,6 +5,18 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [3.1.1] - 2022-04-07
+
+### Added
+
+### Removed
+
+### Changed
+
+### Fixed
+
+- MessagingSession reconnects with refreshed endpoint and credentials if needed
+
```

Comment:
We haven't release 3.1.0 version. I think you should add your CHANGELOG under 3.1.0 section.
And remove 3.1.1 section.

Reply:
@dpwspoon Hi, when do we schedule this release? if it's before the release of 3.1.0. Maybe we should update the version to 3.0.1, if it's a patch version.

Sorry I didn't catch this in the previous comment.

Reply:
@ltrung @devalevenkatesh how do you think?

Reply:
So, this chanelog can be under 3.1.0 we will later fix the changelog once the patch is released. I think I had checked the same with Trung. After release we will need to add 3.0.1 as a patch with this change.

Reply:
The changelog update is a bit of improvement we have to make once we go through this hotfix release once and fix any hiccups if seen. But yes, we would like to add as part of the PR after we release this time.